operator/cmd/mesh/upgrade.go

```diff
@@ -215,7 +215,7 @@ func upgrade(rootArgs *rootArgs, args *upgradeArgs, l clog.Logger) (err error) {
 	}
 	checkUpgradeIOPS(currentProfileIOPSYaml, targetIOPYaml, overrideIOPYaml, l)
 
-	waitForConfirmation(args.skipConfirmation && !rootArgs.dryRun, l)
+	waitForConfirmation(args.skipConfirmation, l)
```

Comment:
```suggestion
	waitForConfirmation(args.skipConfirmation || rootArts.dryRun, l)
```

WDYT? I am not sure it ever makes sense to confirm in dryrun, and it seems to match https://github.com/istio/istio/blob/8032be2664d55b55fd013857a50bb60390b32ec3/operator/cmd/mesh/install.go#L170

Reply:
I think it will make the user confused, what happens if the user wants to execute a dry run with confirmation? I think we also need to refactor the install.go in this case.

so in the end, the user will have 4 options
1. dry-run with confirmation
2. dry-run without confirmation
3. install/upgrade with confirmation
4. install/upgrade without confirmation

Reply:
I think we should be consistent with `install`. There is no need to confirm a dry run - the point of confirmation is to avoid accidentally making changes to your cluster

Reply:
got it, commit your suggestion.